helix-term/src/commands.rs

```diff
@@ -4050,7 +4049,7 @@ fn jump_forward(cx: &mut Context) {
         view.doc = *id;
         let selection = selection.clone();
         let (view, doc) = current!(cx.editor); // refetch doc
-        doc.set_selection(view.id, selection);
+        doc.set_selection(view.id, selection.borrow().clone());
```

Comment:
I'm trying to learn so excuse me if this is obvious. But what effect does `clone().borrow().clone()` give us?

Reply:
The first `clone()` is a clone of the [`Rc`](https://doc.rust-lang.org/std/rc/struct.Rc.html) and is needed to avoid borrowing conflict with `current!`. `Rc` clones are very cheap (clone of the reference, not the contained data) and I believe that one expires at the end of the block.

The call to `borrow()` is [`RefCell::borrow`](https://doc.rust-lang.org/std/cell/struct.RefCell.html#method.borrow) (note: the `Rc` is automatically dereferenced into the contained `RefCell`) which gives a reference to the data contained in the `RefCell`.

Then we need to `clone()` that since we want to pass a new `Selection` to `Document::set_selection` (more borrowing conflict with `current!`).

Reply:
I suppose we can get rid of that Rc clone by moving the borrow part up